src/gpu/amd/sycl_hip_stream.cpp

```diff
@@ -106,17 +106,12 @@ status_t sycl_hip_stream_t::init() {
 
 status_t sycl_hip_stream_t::interop_task(
         std::function<void(::sycl::handler &)> sycl_hip_interop_) {
-    try {
-        auto event = queue().submit([&](::sycl::handler &cgh) {
-            cgh.depends_on(sycl_ctx().get_sycl_deps().events);
-            sycl_hip_interop_(cgh);
-        });
-        this->sycl_ctx().get_sycl_deps().events = {event};
-        return status::success;
-    } catch (std::runtime_error &e) {
-        error::wrap_c_api(status::runtime_error, e.what());
-        return status::runtime_error;
-    }
+    auto event = queue().submit([&](::sycl::handler &cgh) {
+        cgh.depends_on(sycl_ctx().get_sycl_deps().events);
+        sycl_hip_interop_(cgh);
+    });
+    this->sycl_ctx().get_sycl_deps().events = {event};
+    return status::success;
```

Comment:
We should probably keep the original version here but instead of throwing an exception with `error::wrap_c_api` we could just return `status::runtime_error`.

Reply:
Thanks